src/System.Threading.Channels/tests/Stress.cs

```diff
@@ -0,0 +1,423 @@
+            BoundedChannelOptions boundedOptions = options as BoundedChannelOptions;
+            bool shouldReadAllWrittenValues = boundedOptions == null || boundedOptions.FullMode == BoundedChannelFullMode.Wait;
+
+            List<Task> taskList = new List<Task>();
+
+            int readerTasksCount;
+            int writerTasksCount;
+
+            if (options.SingleReader)
+            {
+                readerTasksCount = 1;
+                writerTasksCount = options.SingleWriter ? 1 : MaxTaskCounts - 1;
+            }
+            else if (options.SingleWriter)
+            {
+                writerTasksCount = 1;
+                readerTasksCount = MaxTaskCounts - 1;
+            }
+            else
+            {
+                readerTasksCount = MaxTaskCounts / 2;
+                writerTasksCount = MaxTaskCounts - readerTasksCount;
+            }
+
+            int readCount = 0;
+            int wroteCount = 0;
+
+            for (int i=0; i < readerTasksCount; i++)
+            {
+                taskList.Add(new Task(() => {
+                    while (true)
+                    {
+                        try
+                        {
+                            if (!reader.TryRead(out int value))
+                            {
+                                value = reader.ReadAsync().GetAwaiter().GetResult();
+                            }
+                            Interlocked.Increment(ref readCount);
+                        }
+                        catch (ChannelClosedException)
+                        {
+                            break;
+                        }
+                    }
+                }));
+            }
+
+            int numberToWriteToQueue = -1;
+            for (int i=0; i < writerTasksCount; i++)
+            {
+                taskList.Add(new Task(() => {
+                    while (true)
+                    {
+                        int incrementedValue = Interlocked.Increment(ref numberToWriteToQueue);
+                        if (incrementedValue >= MAX_NUMBER_TO_WRITE_TO_CHANNEL)
+                        {
+                            // we have reached the max we can write.
+                            if (incrementedValue == MAX_NUMBER_TO_WRITE_TO_CHANNEL)
+                            {
+                                while (wroteCount < MAX_NUMBER_TO_WRITE_TO_CHANNEL)
+                                {
+                                    // we have some tasks is currently writing and not finished yet. wait them to finish.
+                                    Task.Delay(100);
+                                }
+
+                                writer.Complete();
+                            }
+                            break;
+                        }
+
+                        if (!writer.TryWrite(incrementedValue))
+                        {
+                            writer.WriteAsync(incrementedValue).GetAwaiter().GetResult();
```

Comment:
What's the goal of this pattern?  Note that for the unbounded channel, WriteAsync will never be used as TryWrite will always succeed.  So if there's a desire in this test to test WriteAsync, that won't happen in such cases.

Reply:
I am trying to write the code that can work with all channel even without the knowledge what type of channel I am using. if you have better pattern suggestion that works with all channels, I am ok to use. note that, I wrote that to work even with the unbuffered channels.